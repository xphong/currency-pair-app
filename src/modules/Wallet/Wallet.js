import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';

export class WalletContainer extends Component {
  static propTypes = {
    balance: PropTypes.object.isRequired,
  };

  render() {
    const { balance } = this.props;

    return (
      <div>
        $$$${balance.dollar}
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    balance: state.wallet.balance
  };
}

export default connect(
  mapStateToProps
)(WalletContainer);
