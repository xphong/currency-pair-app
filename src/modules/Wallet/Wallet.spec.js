import React from 'react';

import { WalletContainer } from './Wallet';

describe('Wallet Container', () => {
  it('should render Wallet', () => {
    const props = {
      balance: {
        USD: 1,
        BTC: 0
      }
    }
    const wrapper = shallow(<WalletContainer {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
