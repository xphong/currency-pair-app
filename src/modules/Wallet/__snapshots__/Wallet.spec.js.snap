// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet Container should render Wallet 1`] = `
<div>
  <Balance
    balance={
      Object {
        "BTC": 0,
        "USD": 1,
      }
    }
  />
</div>
`;
